service/operator/service.go

```diff
@@ -207,7 +212,13 @@ func (s *Service) GetCurrentState(obj interface{}) (interface{}, error) {
 			if err != nil {
 				return nil, microerror.MaskAny(err)
 			}
+			// Ensure that the map is assignable. This prevents panics down the road
+			// in case the config-map has no data at all.
+			if len(k8sConfigMap.Data) == 0 {
+				k8sConfigMap.Data = map[string]string{}
+			}
```

Comment:
We initially set the empty map in `GetCreateState`, but overwrote content sometimes. This is now here to ease the work of making up the create-state. Feels also more right to make up and initialized everything around the current state here in `GetCurrentState`.